### PR TITLE
[DOCS] Value list exceptions for all rule types

### LINKED ISSUE #2471: [DOCS] Value list exceptions for all rule types
## Description

PR https://github.com/elastic/kibana/pull/133254 adds support for value list exceptions to all rule types (previously not supported for event correlation and threshold rule types). 

The dev PR mentions several caveats and details, which I'll just paste here:

> Adds value list exception support to every rule type with a few caveats. Structurally, this PR adds the definition of a "small" list that is able to be included directly in the rule executor's elasticsearch query instead of filtering out large lists in the post-execution process. This allows us to add value list exceptions for every rule type given some constraints, but still keep the post-execution logic and not degrade any current expected functionality. 
> 
> #### Definitions
> 
> - "Small" value lists are defined as Keyword, IP value, or IP range* type lists that are under 65,536 items
> - Small value lists are able to be used for every rule type
> - Large value lists are defined as Keyword, IP value, or IP range* type lists that are over 65,536 items OR any sized text type list
> - Large value lists are only able to be used for Query, Saved Query, ML, and Threat match rule types
> - IP range value lists have two value types: slash notation (127.0.0.1/32) and dash notation (127.0.0.1-127.0.0.4)
> 
> #### Caveats
> 
> - IP range value lists are considered small only if the slash notation values are under 65,536, dash notation values require us to build the exception clauses differently and thus are limited to 200 for a small list size
> - Text type value lists are currently not supported by small value lists

## Additional notes

- Large lists will still display in the drop-down for selecting a value list, but will be grayed out. Might want to explain in docs what that means (UI may also provide info)

### PR DESCRIPTION
**Pull request by joepeeples**
Resolves #2471.

Preview: [Rule exceptions and value lists](https://security-docs_2562.docs-preview.app.elstc.co/guide/en/security/master/detections-ui-exceptions.html) - Mostly updates the "Create value lists" sections

---

Note: Docs PR #2546 (New exceptions UI) is also updating portions of the "Rule exceptions and value lists" page, but I _think_ we'll be able to avoid conflicts since we're working in different areas. 🤞

### CODE REVIEW

File: docs/detections/detections-ui-exceptions.asciidoc [hunk 1]
```diff
@@ -21,15 +21,18 @@ with these types:
...
-IMPORTANT: Operators `is in list` and `is not in list` are not available for
-threshold and event correlation rules.
-
 TIP: You can also use value lists as the <<indicator-value-lists,indicator match index>> when creating an indicator match rule.
 
 [float]
 [[manage-value-lists]]
 == Create value lists
 
+When creating a value list for a rule exception, be mindful of the list's size and data type. In general, value list exceptions are supported for all rule types, but there are some limitations for especially large lists or certain data types. The following value list types can _only_ be used with custom query, machine learning, and indicator match rule types:
```
Comment (jmikell821) [inline docs/detections/detections-ui-exceptions.asciidoc:30]: Flipping this to active voice. 
```suggestion
When creating a value list for a rule exception, be mindful of the list's size and data type. In general, all rule types support value list exceptions, but there are some limitations for especially large lists or certain data types. The following value list types can _only_ be used with custom query, machine learning, and indicator match rule types:
```

File: docs/detections/detections-ui-exceptions.asciidoc [hunk 2]
```diff
@@ -130,8 +133,9 @@ image::images/add-exception-ui.png[]
 +
 [NOTE]
 =======
-*  An exception defined by a value list must use `is in list` or `is not in list` in all conditions.
+* An exception defined by a value list must use `is in list` or `is not in list` in all conditions.
 * Wildcards are not supported in value lists.
+* If a value list can't be used due to <<manage-value-lists,size or data type>>, it will appear unavailable in the *Value* menu.
```
Comment (jmikell821) [inline docs/detections/detections-ui-exceptions.asciidoc:138]:
```suggestion
* If a value list can't be used due to <<manage-value-lists,size or data type>>, it'll be unavailable in the *Value* menu.
```

**Review by jmikell821 — COMMENTED** [review]
Two small nits. LGTM, thanks!

**Review by nastasha-solomon — APPROVED** [review]
LGTM!

**Review by dplumlee — APPROVED** [review]
Looks good!

**Review by benironside — APPROVED** [review]
👍 👍

**Comment by deepikakeshav-qasource** [comment]
Hi @joepeeples,

We have validated the above Kibana documentation with reference to this PR and found that the doc is correctly updated.

image

Hence, we are marking this PR as QA:Approved.

Thank you!

**Pull request merged**